packages/components/modal/src/Modal.tsx

```diff
@@ -49,40 +52,60 @@ const Modal = ({
     return () => {
       window.removeEventListener('keydown', handleKeyDown);
     };
-  }, [onClose]);
+  }, [onClose, isOpen]);
+
+  const handleOverlayClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (
+      closeOnOutsideClick &&
+      modalRef.current &&
+      !modalRef.current.contains(event.target as Node)
+    ) {
+      onClose();
+    }
+  };
 
   return (
     <>
       {isOpen && (
-        <ModalContainerStyled size={size} {...props}>
-          <ModalContentStyled
-            styled={mergedThemeValues as Required<ModalStyledProps>}
-          >
-            {headerTitle != undefined && (
-              <ModalHeader
-                className="modal-header"
-                size={size}
-                styled={styledHeader}
-                headerTitle={headerTitle}
-                onClose={onClose}
-              />
-            )}
-
-            <ModalBodyStyled
-              className="modal-body"
-              size={size}
+        <ModalContetWrapperStyled size={size} onClick={handleOverlayClick}>
+          <ModalContainerStyled size={size} {...props}>
+            <ModalContentStyled
+              ref={modalRef}
               styled={mergedThemeValues as Required<ModalStyledProps>}
+              onClick={(e: React.MouseEvent<HTMLDivElement>) =>
+                e.stopPropagation()
+              }
             >
-              {childrenBody}
-            </ModalBodyStyled>
+              {headerTitle != undefined && (
```

Comment:
Read more if you want to understand with the use of typeof:
https://builtin.com/software-engineering-perspectives/javascript-array-typeof

```suggestion
              {typeof headerTitle !== 'undefined' && (
```